src/stdx.zig

```diff
@@ -815,6 +815,36 @@ test fmt_int_size_bin_exact {
     });
 }
 
+
+/// Like std.fmt.bufPrint, but checks, at compile time, that the buffer is sufficiently large.
+pub fn array_print(
+    comptime n: usize,
+    buffer: *[n]u8,
```

Comment:
Could pass an `anytype` here,  but figured that with `*[n]` the signature would be more understandable. 

Alternatively, instead of taking an array as an argument, this could _return_ a bounded array of the precise size, but I think it's actually beneficial to see a hard-coded constant at the call-site, to serve as a sanity check that we aren't accidentally allocating 4megs on the stack.